enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,566 @@
+last-updated: 2023-06-02
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-1140
+see-also: []
+---
+
+# CSI Snapshotting Integration
+
+## Summary
+
+MicroShift is a small form-factor, single-node OpenShift targeting IoT and Edge Computing use cases characterized 
+by tight resource constraints, unpredictable network connectivity, and single-tenant workloads. See 
+[kubernetes-for-devices-edge.md](./kubernetes-for-device-edge.md) for more detail.
+
+This document proposes the integration of the CSI Snapshot Controller to support backup and restore scenarios for 
+cluster workloads.  The snapshot controller, along with the CSI external snapshot sidecar, will provide an API 
+driven pattern for managing stateful workload data.
+
+## Motivation
+
+CSI snapshot functionality was originally excluded from the CSI driver integration in MicroShift to support the 
+low-resource overhead goals of the project. However, user feedback has made it clear that a supportable, robust 
+backup/restore solution is necessary.  While it would be possible to run a workflow out-of-band to manage workload 
+data, this would be reinventing the wheel and contribute significantly to technical debt.  CSI Snapshots are already 
+integrated into Openshift with ongoing downstream support. 
+
+### User Stories
+
+As an edge device owner running stateful workloads on MicroShift, I want to create in-cluster snapshots of that state 
+and to restore workloads to that state utilizing existing Kubernetes patterns.
+
+### Goals
+
+* Enable an in-cluster workflow for snapshotting and restoration of cluster workload data
+* Follow the [MicroShift design principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* Enable backup/restore workflows external to MicroShift
+* Provide a workflow for exporting data from a MicroShift cluster
+* Provide a means of opting out.  This falls under platform composability and is out of scope
+
+## Proposal
+
+Deploy the CSI snapshot controller and CSI plugin sidecar with MicroShift out of the box to provide users with a means of 
+snapshotting, cloning, and restoring workload state.
+
+### Workflow Description
+
+#### Deploy MicroShift with Snapshotting Support
+
+_Prerequisites_
+
+* The device owner has created an LVM volume group and a thin-pool on this volume group.
+* The `/etc/microshift/lvmd.yaml` config includes a `deviceClass` to represent the thin-pool.  The lvmd.yaml may 
+contain a mix of thin and thick `deviceClasses`.
+
+```yaml
+device-classes:
+- name: ssd-thin
+  volume-group: myvg1
+  type: thin
+  thin-pool:
+    name: pool0
+    overprovision-ratio: 50.0
+- name: default
+  volume-group: rhel
+  default: true
+  spare-gb: 0
+```
+
+_Workflow_
+
+1. The user installs the MicroShift RPMs, either via os-rpmtree layer or package manager.
+2. The user starts the MicroShift systemd service
+3. MicroShift starts the Service Manager loop to begin deploying cluster resources.
```

Comment:
I think steps 1-to-3 can be collapsed into a single one as installation related aspects are not relevant for this proposal. Let's just describe what happens when MicroShift service starts?